src/CMakeLists.txt

```diff
@@ -3,11 +3,12 @@ target_compile_options(secutils_compilation_options INTERFACE
     -Wall -Woverflow -Wextra -Wswitch -Wmissing-prototypes -Wstrict-prototypes
     -Wformat -Wformat-security -Wtype-limits -Wundef
     -Wsign-compare -Wpointer-arith -Wunused-parameter
-    -pedantic -Werror
+    -pedantic
+    # -Werror is enabled only for development and CI using Makefile_v1 without NDEBUG
```

Comment:
Why that?

Reply:
I want to do this for the genCMPClient and libcmp and propose to do this also for libSecUilts
due to experience of users trying to build the libraries using newer/other compilers (e.g., from other OS variants such as Mac).
These tools can be more picky/pedantic on minor potential issues than the compilers typically used.
The idea is to let such users get past small issues that may lead to unexpected warnings, without getting the build aborted.
They will still see the warnings and then can decide for themselves to ignore them or not.

Anyway, the CI runs continue using `-Werror` with `Makefile_v1` such that all warnings emitted
by the typical compilers lead to CI failure and in this way will bring all compiler-reported issues to the maintainers' attention.

Yet if you feel uncomfortable with this change, I can 
* undo it for libSecUtils in general or
* undo it for the case that CMake is used with debug builds (such that `-Werror` would be kept for debug builds both using `Makefile_v1` and CMake) - 

as you prefer.